### PR TITLE
Improving documentation for Redis Stream Source

### PR DESCRIPTION
**Pull request by aavarghese**
Signed-off-by: Ansu Varghese <avarghese@us.ibm.com>

Fixes #

## Proposed Changes

-
-
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: source/README.md [hunk 1]
```diff
@@ -1,74 +1,113 @@
-# Knative Redis Source
+# Redis Stream Source
+
+The Redis Stream Event Source for Knative reads messages from a Redis Stream and
+sends them as CloudEvents to the referenced Sink, which can be a Kuberentes service
```
Comment (lionelvillard) [inline source/README.md:4]: nit: Kuberentes -> Kubernetes

File: source/README.md [hunk 2]
```diff
@@ -1,74 +1,113 @@
...
+name and consumer [`group`](config/300-redisstreamsource.yaml) name will be 
+created by the receive adapter, if they don't already exist.
+
+The number of consumers in the consumer group can also be configured via data in 
+[`config-redis`](config/config-redis.yaml). This makes it possible for each consumer to
+consume different messages arriving in the stream. Each consumer has an unique consumer 
+name which is a string created by the receive adapter.
+
+When a Redis Stream Source resource is deleted, all the consumers in the group
+are gracefully shut down/deleted, before the consumer group itself is destroyed.
```
Comment (lionelvillard) [inline source/README.md:20]: shut down-> shutdown

File: source/README.md [hunk 3]
```diff
@@ -92,14 +131,55 @@ Data,
...
 kubectl delete ns redex
 ```
 
-## Release Notes
+## Reference
+
+### Prerequisites
+
+* A Redis installation. Instructions to deploy a local Redis are above. 
+A cloud based Redis can be installed on IBM Cloud (for example), at https://cloud.ibm.com/catalog/services/databases-for-redis
```
Comment (lionelvillard) [inline source/README.md:146]: I would remove this line.

Comment (aavarghese) [reply source/README.md]: Done (687abaa)

File: source/README.md (hunk 2 above)
Comment (aavarghese) [reply source/README.md]: Done (687abaa)

File: source/README.md (hunk 1 above)
Comment (aavarghese) [reply source/README.md]: Done (687abaa)

File: source/pkg/reconciler/streamsource/resources/receive_adapter_test.go [hunk 4]
```diff
@@ -35,7 +35,7 @@ func TestMakeReceiveAdapter(t *testing.T) {
 		},
```
Comment (lionelvillard) [inline source/pkg/reconciler/streamsource/resources/receive_adapter_test.go:35]: this file does not seem related to the PR.

Comment (aavarghese) [reply source/pkg/reconciler/streamsource/resources/receive_adapter_test.go]: Removed.
